packages/react/src/components/Table/TableHeader.tsx

```diff
@@ -1,21 +1,64 @@
-import React, { ThHTMLAttributes } from 'react';
+import React, { useRef, useEffect, ThHTMLAttributes } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import Icon from '../Icon';
+
+// these match aria-sort's values
+type SortDirection = 'ascending' | 'descending' | 'none';
 
 interface TableHeaderProps extends ThHTMLAttributes<HTMLTableCellElement> {
   children: React.ReactNode;
+  sortDirection?: SortDirection;
+  onSort?: () => void;
   className?: string;
 }
 
-const TableHeader = ({ children, className, ...other }: TableHeaderProps) => (
-  <th className={classNames('TableHeader', className)} {...other}>
-    {children}
-  </th>
-);
+const TableHeader = ({
+  children,
+  sortDirection,
+  onSort,
+  className,
+  ...other
+}: TableHeaderProps) => {
+  const sortButtonRef = useRef<HTMLButtonElement>(null);
+  useEffect(() => {
+    if (!sortDirection || sortDirection === 'none') {
+      return;
+    }
+    sortButtonRef.current?.focus();
+  }, [sortDirection]);
+  return (
+    <th
+      aria-sort={sortDirection}
+      className={classNames('TableHeader', className, {
+        'TableHeader--sorting': sortDirection && sortDirection !== 'none'
+      })}
+      {...other}
+    >
+      {onSort && sortDirection ? (
+        <button ref={sortButtonRef} onClick={onSort} className="sort-button">
+          {children}
+          <span>
```

Comment:
Minor nitpick: We should be able to add `aria-hidden="true"` here
```suggestion
          <span aria-hidden="true">
```